pkg/ioutils/readers.go

```diff
@@ -44,18 +47,34 @@ func NewReaderErrWrapper(r io.Reader, closer func()) io.Reader {
 
 type bufReader struct {
```

Comment:
Unrelated to this PR: this should contain some commentary on why `bufio.Reader`, from the stdlib, is insufficient.